requests_hardened/client.py

```diff
@@ -30,9 +30,10 @@ def send(self, request: PreparedRequest, **kwargs: Any) -> Response:
 
     def prepare_request(self, request: Request) -> PreparedRequest:
         url = request.url
-
         if self._config.ip_filter_enable is True:
             headers = request.headers or {}
+            if self._config.user_agent_override:
```

Comment:
`None` should probably be the only case where the override is disabled. An empty user-agent is valid and will be accepted by `requests` library

```suggestion
            if self._config.user_agent_override is not None:
```